SwrveSDKTest/src/test/java/com/swrve/sdk/ConversationActivityTest.java

```diff
@@ -0,0 +1,194 @@
+        assertEquals(1, swrveSpy.campaigns.get(0).getImpressions());
+    }
+
+    @Test
+    public void testConversationWidths() throws Exception {
+        SwrveTestUtils.loadCampaignsFromFile(mActivity, swrveSpy, "conversation_campaign.json", "8d4f969706e6bf2aa344d6690496ecfdefc89f1f");
+        assertNotNull(swrveSpy.campaigns);
+        SwrveConversation conversation = swrveSpy.getConversationForEvent("swrve.messages.showatsessionstart", new HashMap<String, String>());
+        assertNotNull(conversation);
+
+        ActivityController<ConversationActivity> conversationActivityController = Robolectric.buildActivity(ConversationActivity.class);
+        Intent intent = new Intent(RuntimeEnvironment.application, ConversationActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        intent.putExtra("conversation", conversation);
+        ConversationActivity conversationActivity = conversationActivityController.withIntent(intent).create().start().visible().get();
+        assertNotNull(conversationActivity);
+
+        ConversationFragment fragment = conversationActivity.getConversationFragment();
+        assertTrue(fragment.getView() instanceof ConversationRelativeLayout);
+        ConversationRelativeLayout conversationRelativeLayout = (ConversationRelativeLayout)fragment.getView();
+
+        ViewGroup.LayoutParams outsideParams = conversationRelativeLayout.getLayoutParams();
+        assertEquals(ViewGroup.LayoutParams.MATCH_PARENT, outsideParams.width);
+        assertEquals(ViewGroup.LayoutParams.MATCH_PARENT, outsideParams.height);
+
+        View conversationLayoutModal = conversationRelativeLayout.findViewById(com.swrve.sdk.conversations.R.id.swrve__conversation_modal);
+        RelativeLayout.LayoutParams modalParams = (RelativeLayout.LayoutParams) conversationLayoutModal.getLayoutParams();
+        assertEquals(0, modalParams.topMargin);
+        assertEquals(0, modalParams.bottomMargin);
+
+        // increase the width to over the max and refresh the layout
+        int maxModalWidthPx = mActivity.getResources().getDimensionPixelSize(com.swrve.sdk.conversations.R.dimen.swrve__conversation_max_modal_width);
+        outsideParams.width = maxModalWidthPx + 1;
+        conversationRelativeLayout.requestLayout();
+
+        conversationLayoutModal = conversationRelativeLayout.findViewById(com.swrve.sdk.conversations.R.id.swrve__conversation_modal);
+        modalParams = (RelativeLayout.LayoutParams) conversationLayoutModal.getLayoutParams();
+        int topBottomPaddingPx = mActivity.getResources().getDimensionPixelSize(com.swrve.sdk.conversations.R.dimen.swrve__conversation_min_modal_top_bottom_padding);
+        assertEquals(topBottomPaddingPx, modalParams.topMargin);
+        assertEquals(topBottomPaddingPx, modalParams.bottomMargin);
+        assertEquals(365, modalParams.width);
+        assertEquals(ViewGroup.LayoutParams.WRAP_CONTENT, modalParams.height);
```

Comment:
~ ~ Any way you can assert that the height ends up being what you expect it to be with that JSON?